.github/workflows/execute-tests.yml

```diff
@@ -60,14 +60,22 @@ jobs:
       # very slow. Especially any client/server tests seems to be problematic.
       # This is a simple attempt to re-run the tests up to three times if they
       # fail. Does not add any execution time if successful.
-      - name: Run tests attempt 1
-        run: make test
-      - name: Run tests attempt 2
-        if: ${{ failure() }}
-        run: make test
-      - name: Run tests attempt 3
-        if: ${{ failure() }}
-        run: make test
+      - uses: Wandalen/wretry.action@v1.0.20
+        name: Run tests
+        with:
+          command: make test
+          current_path: ${{ inputs.workpath }}
+          attempt_limit: 3
```

Comment:
I suggest rerunning tests at the runner-level rather than restarting the whole suite.

https://pypi.org/project/pytest-rerunfailures/ could be added as a dependency in the future.